rules/reports/reports.rules

```diff
@@ -0,0 +1,72 @@
+# -*- mode: Snakemake -*-
+#
+# ReportGeneration rules
+
+import pandas
+from io import StringIO
+
+rule all_reports:
+    input:
+        TARGET_REPORT
+
+def parse_trim_summary(filename):
+    with open(filename) as f:
+        for line in f:
+            if line.startswith('Input Read Pairs'):
+                vals = re.findall('\D+\: (\d+)', line)
+                keys = ('input', 'both_kept','fwd_only','rev_only','dropped')
+                return(dict(zip(keys, vals)))
+
+def parse_decontam_json(jsonfile):
+    with open(jsonfile) as f:
+        data = json.load(f)
+    return(data['data'])
+
+rule preprocess_report:
+    """Combines the information from multiple preprocessing steps"""
+    input:
+        trim_files = expand(str(QC_FP/'log'/'trimmomatic'/'{sample}.out'), sample=Samples.keys()),
+        json_files = expand(str(QC_FP/'log'/'decontam-human'/'{sample}_summary.json'), sample=Samples.keys())
+    output:
+        str(QC_FP/'preprocess_summary.tsv')
+    run:
+        summary_list = []
+        for i in range(len(input.trim_files)):
```

Comment:
Instead of using a numerical index, suggest zip'ing the lists:

    for trim_file, decontam_file in zip(input.trim_files, input.json_files):
        tname = ...